### PR TITLE
Documentation fix in contributing.md file

### PR DESCRIPTION
**Pull request by Cyril-Anto**
Fixes #104872 Errors in contributing.md file

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1064,23 +1064,24 @@ To run clang-tidy locally, follow these steps:
...
-make clang-tidy
-# Run clang-tidy only on your changes
-make clang-tidy CHANGED_ONLY=--changed-only
+# Run lintrunner on the entire codebase
+lintrunner
+# Run lintrunner on your changes only
+lintrunner CHANGED_ONLY=--changed-only
 ```
-This internally invokes our driver script and closely mimics how clang-tidy is run on CI.
+This will run clang-tidy on the entire codebase or on your changes only, respectively.
```
Comment (kit1980) [inline CONTRIBUTING.md:1081]: lintrunner will not only run clang-tidy, but other linters.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1064,23 +1064,24 @@ To run clang-tidy locally, follow these steps:
 1. Install clang-tidy.
 We provide custom built binaries which have additional checks enabled. You can install it by running:
 ```bash
-python3 -m tools.linter.clang_tidy.generate_build_files
+pip3 install clang-tidy
 ```
 We currently only support Linux and MacOS (x86).
```
Comment (malfet) [inline CONTRIBUTING.md:1069]: This steps becomes redundant, if you are to use `lintrunner`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1064,23 +1064,24 @@ To run clang-tidy locally, follow these steps:
...
-
-3. Run clang-tidy
+3. Run lintrunner
 ```bash
-# Run clang-tidy on the entire codebase
-make clang-tidy
-# Run clang-tidy only on your changes
-make clang-tidy CHANGED_ONLY=--changed-only
+# Run lintrunner on the entire codebase
+lintrunner
```
Comment (malfet) [inline CONTRIBUTING.md:1077]:
```suggestion
lintrunner --take CLANG-TIDY --all-files
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1064,23 +1064,24 @@ To run clang-tidy locally, follow these steps:
...
+# Run lintrunner on your changes only
+lintrunner CHANGED_ONLY=--changed-only
```
Comment (malfet) [inline CONTRIBUTING.md:1079]:
```suggestion
lintrunner --take CLANGTIDY
```

**Review by malfet — CHANGES_REQUESTED** [review]
Please see suggested changes.
And please test the commands, read tool documentation before proposing updates (i.e. `lintrunner CHANGED_ONLY=--changed-only` is just syntactically incorrect)

**Comment by Cyril-Anto** [comment]
@malfet Please review the suggested changes.

**Comment by Cyril-Anto** [comment]
@eellison Can you please review the suggested changes

**Comment by Cyril-Anto** [comment]
@malfet I've made the changes based on your suggestions, and can you please review them again?

**Comment by malfet** [comment]
@Cyril-Anto did you tried running the commands you've proposed?(Hint: they will not work, as you forgot to init the linters) And  last paragraph contradicts proposed linter invocation, isn't it?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1061,26 +1061,26 @@ for the simple commands we use for this.
...
+lintrunner --take clang-tidy --all-files
+# Run lintrunner on your changes only
+lintrunner --take clang-tidy --changed-only
 ```
-This internally invokes our driver script and closely mimics how clang-tidy is run on CI.
+
+This will run a variety of linters on the entire codebase or on your changes only, respectively.
+One of the linters that lintrunner runs is clang-tidy.
+
+The reason why LLVM 8 is needed is because it is the version of LLVM that clang-tidy was tested with.
```
Comment (malfet) [inline CONTRIBUTING.md:1081]: Sorry, can you please elaborate what you mean by that? As one can see from tools/linter/adapters/s3_init_config.json as of time of this writing, clang-tidy from llvm-15 is used

**Review by malfet — CHANGES_REQUESTED** [review]
Please do not request re-reviews until you address the feedback

**Pull request closed without merging**